pyproj/_transformer.pyx

```diff
@@ -330,14 +330,15 @@ cdef PJ* proj_create_crs_to_crs(
         ELSE:
             raise NotImplementedError("only_best requires PROJ 9.2+.")
 
-
-    cdef PJ* transform = proj_create_crs_to_crs_from_pj(
-        ctx,
-        source_crs,
-        target_crs,
-        area,
-        options,
-    )
+    cdef PJ* transform = NULL
+    with nogil:
+        transform = proj_create_crs_to_crs_from_pj(
```

Comment:
Just for my own edification, does the declaration need to be moved outside the `with nogil` because Cython doesn't like it?

Reply:
Precisely. cdefs are only allowed directly within a function and not within a `with` or `try` block.